apps/aechannel/src/aesc_fsm.erl

```diff
@@ -1541,6 +1571,30 @@ new_onchain_tx(channel_close_solo_tx, Opts,
                , ttl        => TTL},
     {ok, Tx} = new_onchain_tx_(aesc_close_solo_tx, Opts1, CurrHeight),
     {ok, Tx, []};
+new_onchain_tx(channel_snapshot_solo_tx, Opts,
+               #data{ on_chain_id = ChanId
+                    , state       = State } = D, CurrHeight) ->
+    Account = my_account(D),
+    TTL = adjust_ttl(maps:get(ttl, Opts, 0)),
+    {Round, SignedTx} = aesc_offchain_state:get_latest_signed_tx(State),
+    case aesc_utils:is_offchain_tx(SignedTx) of
+        true ->
+            case later_round_than_onchain(ChanId, Round) of
+                true ->
+                    Payload = aetx_sign:serialize_to_binary(SignedTx),
+                    Opts1 =
+                        Opts#{ channel_id => aeser_id:create(channel, ChanId)
+                             , from_id    => aeser_id:create(account, Account)
+                             , payload    => Payload
+                             , ttl        => TTL },
+                    {ok, Tx} = new_onchain_tx_(aesc_snapshot_solo_tx, Opts1, CurrHeight),
+                    {ok, Tx, []};
+                false ->
+                    {error, already_onchain}
```

Comment:
Shouldn't we pass this error to the client?

Reply:
I think we do, although there is no error code attached to this particular error reason.

Reply:
Error code and description added, and test case enhanced to verify error returns.